apps/discord/auth.py

```
@@ -0,0 +1,42 @@
+
+        if not signature or not timestamp:
+            raise HTTPException(status_code=401, detail="Missing signature headers")
+
+        # Create message for verification
+        message = timestamp.encode() + body
+
+        try:
+            verify_key.verify(message, bytes.fromhex(signature))
+        except BadSignatureError:
+            # Either an unauthorized request or Discord's "negative control" check
+            raise HTTPException(status_code=401, detail="Invalid request")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)